operators/src/pro/cache/tile_cache.rs

```diff
@@ -29,16 +30,425 @@ pub struct TileCacheBackend {
+
+        trace!(
+            "Inserted tile for query {} into landing zone. Landing zone size: {}. Landing zone size used: {}. Landing zone used percentage: {}",
+            query_id, self.landing_zone_size.total_byte_size(), self.landing_zone_size.byte_size_used(), self.landing_zone_size.size_used_fraction()
+        );
+
+        Ok(())
+    }
+
+    /// Finish the query and make the tiles available in the cache
+    ///
+    /// # Panics
+    ///
+    /// Panics if there is aleady a cache entry for the query. This should never happen.
+    ///
+    pub fn finish_inserting_query(
+        &mut self,
+        key: &CanonicOperatorName,
+        query_id: &QueryId,
+    ) -> Result<(), CacheError> {
+        // TODO: maybe check if this cache result is already in the cache or could displace another one
+
+        // this should always work, because the query was inserted at some point and then the cache entry was created
+        let (landing_zone, landing_zone_size) = self
+            .landing_zone_and_size_mut(key)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        let active_query = landing_zone
+            .remove(query_id)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        // remove the size of the query from the landing zone
+        landing_zone_size.remove_element_bytes(query_id);
+        landing_zone_size.remove_element_bytes(&active_query);
+
+        // debug output
+        debug!(
+            "Finished query {}. Landing zone size: {}. Landing zone size used: {}, Landing zone used percentage: {}.",
+            query_id, self.landing_zone_size.total_byte_size(), self.landing_zone_size.byte_size_used(), self.landing_zone_size.size_used_fraction()
+        );
+
+        // move entry from landing zone into cache
+        let cache_entry: CacheEntry = active_query.into();
+        let cache_entry_id = CacheEntryId::new();
+
+        // calculate size of cache entry. This might be different from the size of the landing zone entry.
+        let cache_entry_size_bytes = cache_entry.byte_size() + cache_entry_id.byte_size();
+
+        let (cache, cache_size, cache_lru) = self
+            .cache_and_size_and_lru_mut(key)
+            .expect("There must be cache elements entry if there was a loading zone");
+
+        let old_cache_entry = cache.insert(cache_entry_id, cache_entry);
+        let old_lru_entry = cache_lru.push(cache_entry_id, key.clone());
+        assert!(old_lru_entry.is_none()); // this should always work, because we just inserted a new CacheEntryId
+        assert!(old_cache_entry.is_none()); // this should always work, because we just inserted a new CacheEntryId
+
+        // cache bound can be temporarily exceeded as the entry is moved form the landing zone into the cache
+        // but the total of cache + landing zone is still below the bound
+        cache_size.add_bytes_allow_overflow(cache_entry_size_bytes);
```

Comment:
you could, instead, have a method like `self.evict_elements_to_free_n_bytes(cache_entry_size_bytes)` and then add the `cache_entry_size_bytes` regularly.

Otherwise: is overflow the right notion here? maybe just call it unchecked? or without_bound_check?